test/unittests/tools/gnu/test_gnutoolchain.py

```diff
@@ -235,3 +235,24 @@ def test_crossbuild_to_android(build_env_mock):
     assert gnutc.triplets_info["host"]["triplet"] == "aarch64-linux-android"
     assert env_vars["LD"] == os.path.join(ndk_bin, "ld")  # exists
     assert env_vars["STRIP"] == os.path.join(ndk_bin, "llvm-strip")  # does not exist but appears
+
+
+def test_gnu_toolchain_conf_configure_args():
+    """ Validate that tools.gnu:configure_args are passed to the configure_args when
+        building with GnuToolchain.
+        The configure args should be passed as a list-like object.
+    """
+    conanfile = ConanFileMock()
+    conanfile.settings = MockSettings({"os": "Linux", "arch": "x86_64"})
+    conanfile.conf = Conf()
+    conanfile.conf.define("tools.gnu:configure_args", ["--foo", "--bar"])
+
+    tc = GnuToolchain(conanfile)
+    assert tc.configure_args["--foo"] is None
+    assert tc.configure_args["--bar"] is None
+
+    conanfile.conf.define("tools.gnu:configure_args", "--foo --bar")
+    with pytest.raises(ConanException) as expected:
+        GnuToolchain(conanfile)
+    assert "[conf] tools.gnu:configure_args must be a list-like object. " \
+           "The value '--foo --bar' introduced is a str object" in str(expected)
```

Comment:
I think that we can remove this check as it's part of the `conf.get(..., check_type="")` behavior:

```suggestion
```